test/test_type_promotion.py

```diff
@@ -686,6 +691,94 @@ def test_integer_div_deprecated(self, device, dtype):
         with self.maybeWarnsRegex(UserWarning, '^Integer division.+is deprecated.+'):
             torch.addcdiv(a, b, b, out=o)
 
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @float_double_default_dtype
+    @onlyCPU
+    def test_numpy_array_binary_ufunc_promotion(self, device):
+        import operator
+        np_types = numpy_to_torch_dtype_dict.keys()
+        torch_types = numpy_to_torch_dtype_dict.values()
+
+        for np_type, torch_type in itertools.product(np_types, torch_types):
+            t0 = torch.tensor((1,), device=device, dtype=torch_type)
+            t1 = torch.tensor((1,), device=device, dtype=numpy_to_torch_dtype_dict[np_type])
```

Comment:
nit: these names could be better.  Like this could be: "np_as_t" or something.

Reply:
Sure. I'll update them.